core/src/main/java/org/neo4j/ogm/context/MappingContext.java

```diff
@@ -346,10 +346,10 @@ private void removeType(Class type, Long id) {
         typeRegister.remove(metaData, type, id);
     }
 
-    private void remember(Object entity) {
+    private void remember(Object entity, boolean overwrite) {
         ClassInfo classInfo = metaData.classInfo(entity);
         Long id = (Long) EntityAccessManager.getIdentityPropertyReader(classInfo).readProperty(entity);
-        objectMemo.remember(id, entity, classInfo);
+        objectMemo.remember(id, entity, classInfo, overwrite);
```

Comment:
Perhaps `remember` is now doing too much? It seems that it always needs to be called to update labels, but not always to update the object hash.